code/modules/projectiles/projectile.dm

```diff
@@ -67,10 +67,26 @@
 	qdel(src)
 
 /obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)
+	var/turf/target_loca = get_turf(target)
 	if(!isliving(target))
 		return 0
 	var/mob/living/L = target
 	if(blocked < 100) // not completely blocked
+		if(damage && L.blood_volume && damage_type == BRUTE)
+			var/splatter_dir = dir
+			if(starting)
+				splatter_dir = get_dir(starting, target_loca)

```

Comment:
@Fox-McCloud I suggest adding `target_loca = get_step(target_loca, splatter_dir)` below this line to have the blood splatter a tile behind the mob in the direction of the splatter as the graphic suggests the blood would land back there.

This works for walls and windows and diagonals.

![bloodflung](https://user-images.githubusercontent.com/12377767/28243610-c0476636-699f-11e7-8ae0-b4925f63ebbf.PNG)